File: Two_Pointers/Patterns/Pattern6/Pattern6.java
class Pattern6{
    public static void main(String[] args) {
        String s="abcabc";
        int[] check=new int[3];
        int count=0;
        int left=0;
        int right=0;

        while(right<s.length()){
            char c=s.charAt(right);
            check[c-'a']++;
            while(check[0]>0 && check[1]>0 && check[2]>0){
                count+=(s.length()-(right));
                char d=s.charAt(left);
                check[d-'a']--;
                left++;
            }
            right++;
        }
    System.out.println(count);
    }
}

File: Two_Pointers/Patterns/Pattern6/Pattern6.py
s="abcabcc"
left=0
right=0
check=[0]*3
count=0

while right<len(s):

    a1=s[right]
    a2=ord(a1)-ord('a')
    check[a2]+=1

    while check[0]>0 and check[1]>0 and check[2]>0:
        b1=s[left]
        b2=ord(b1)-ord('a')
        check[b2]-=1
        count=count+len(s)-right
        left+=1
    right+=1
print(count)
